src/status.rs

```diff
@@ -378,7 +378,7 @@ impl Status {
                             "modified:" => DiffType::Modified,
                             "renamed:" => DiffType::Renamed,
                             "deleted:" => DiffType::Deleted,
-                            _ => panic!("Unknown prefix: `{prefix}`"),
+                            _ => return Err(Error::msg("Unknown prefix: `{prefix}`")),
```

Comment:
The string here will not actually be formatted with the value of `prefix`. You can use the `format!` macro to fix this:
```suggestion
                            _ => return Err(Error::msg(format!("Unknown prefix: `{prefix}`"))),
```
This also applies to the other case